### PR TITLE
master-qa-60285-5 MFA poc source with new automated test

### PR DESCRIPTION
**Pull request by gaborlovasdotcom**
https://issues.liferay.com/browse/LRQA-60285

### CODE REVIEW

File: modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/java/com/liferay/multi/factor/authentication/poc/internal/checker/PocMFAChecker.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
+/**
```
Comment (liferay-appsec) [inline modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/java/com/liferay/multi/factor/authentication/poc/internal/checker/PocMFAChecker.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/java/com/liferay/multi/factor/authentication/poc/internal/settings/definition/MFAPocCompanyConfigurationBeanDeclaration.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (liferay-appsec) [inline modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/java/com/liferay/multi/factor/authentication/poc/internal/settings/definition/MFAPocCompanyConfigurationBeanDeclaration.java:1]: When introducing a new class that implements an interface
        take into account:

* The class name should end with the interface name
* The package of the class matches the package of the interface



Refer to documentation:

File: modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/resources/META-INF/resources/mfa_poc_checker/verify_browser.jsp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<h3>
+	<liferay-ui:message key="please-enter-something" />:
+</h3>
+
+<aui:input label="mfa-poc" name="mfaPoc" showRequiredLabel="yes" />
```
Comment (liferay-appsec) [inline modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/resources/META-INF/resources/mfa_poc_checker/verify_browser.jsp:23]: This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation:

File: modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+com.liferay.multi.factor.authentication.poc.internal.checker.PocMFAChecker=Multi-Factor Authentication POC
```
Comment (liferay-appsec) [inline modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-poc/src/main/resources/content/Language.properties:1]: Check that added properties are either titles or sentences

**Review by liferay-appsec — COMMENTED** [review]
This is an automatic review suggestion

**Comment by csierra** [comment]
please see https://github.com/gaborlovasdotcom/liferay-portal/pull/56

**Pull request closed without merging**